src/realm/decimal128.cpp

```diff
@@ -457,6 +492,18 @@ Decimal128& Decimal128::operator+=(Decimal128 rhs)
     return *this;
 }
 
+Decimal128& Decimal128::operator-=(Decimal128 rhs)
+{
+    unsigned flags = 0;
+    BID_UINT128 x = to_BID_UINT128(*this);
+    BID_UINT128 y = to_BID_UINT128(rhs);
+
+    BID_UINT128 res;
+    bid128_sub(&res, &x, &y, &flags);
```

Comment:
Can this be done without memcpy?

Reply:
To be honest I don't know - I just copied the `operator+=` implementation. However since the `bid128_*` APIs take pointers to `BID_UINT128` then perhaps we can avoid the copy by `reinterpret_cast`-ing the `Bid128` `m_value` in `Decimal128`. I just don't know if we can do that, given that the internal layout of `BID_UINT128` is unclear to me.

Reply:
According to @jedelbo we can't just `reinterpret_cast` away, unfortunately.